controller.go

```diff
@@ -106,6 +111,7 @@ func New() (NetworkController, error) {
 	if err := initDrivers(c); err != nil {
 		return nil, err
 	}
+	c.handler = newHTTPHandler(c)
```

Comment:
For modularity sake, I would prefer the api.go under `api` package and the `api` package can depend on the libnetwork. But not the other way around.

Hence, this initialization piece must be ideally under the `api` package and the libnetwork core should not have any direct dependancies on the API itself.